### PR TITLE
Template render richtext, allows template override

### PR DESCRIPTION
**Pull request by allcaps**
Resolves https://github.com/wagtail/wagtail/issues/1214

* Do the tests still pass? YES
* Does the code comply with the style guide? YES
* For Python changes: Have you added tests to cover the new/fixed behaviour? YES
* For front-end changes: Did you test on all of Wagtail’s supported browsers? N/A
* For new features: Has the documentation been updated accordingly? YES

### CODE REVIEW

File: wagtail/core/templates/wagtailcore/shared/richtext.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<div class="rich-text">{{ html|safe }} LALA</div>
```
Comment (zerolab) [inline wagtail/core/templates/wagtailcore/shared/richtext.html:1]: I love your test content :)

` LALA` makes the tests fail.

Other than that, the approach here is 🔥 for not adding yet another setting.

**Comment by gasman** [comment]
Nice one @allcaps!
I think we should make a plan for dropping the `<div>` in the default configuration - enough people have complained about it that it wouldn't be right to leave it as a recipe that developers have to specifically seek out.

I'd be OK with making that change in a single release as we did for responsive embeds, rather than going through the full multi-release deprecation - in this case, that would mean removing the div from Wagtail's version of the template. If we do that, though, we should really make it as easy as possible for people to revert to the old behaviour, and getting people to set up a template override manually is probably too much of a hoop to jump through. So perhaps we should ship a contrib app that just contains a template override with the div in it, and tell people to add that to their installed apps if they want to keep the old behaviour?

(I have something similar in mind for when we drop SiteMiddleware, moving it to something like `wagtail.contrib.legacy.sitemiddleware` to give people an extra nudge to migrate away from it - although in this case there's less reason to actively discourage people from using it.)

**Comment by allcaps** [comment]
This PR is updated. The suggestions from @gasman are applied:
- Default `richtext` filter does _not_ render the wrapper.
- Add `wagtail.contrib.legacy.richtext`to `INSTALLED_APPS` before `wagtail.core` to get the legacy markup.
- Tests are added.
- Docs are added.
- Tests pass.

File: docs/reference/contrib/index.rst [hunk 2]
```diff
@@ -72,3 +73,9 @@ Provides a TableBlock for adding HTML tables to pages.
 -----------------------
 
 Provides a way to manage redirects.
+
+
+:doc:`legacy_richtext`
+-----------------------
+
+Provides legacy richtext wrapper.
```
Comment (zerolab) [inline docs/reference/contrib/index.rst:81]:
```suggestion
Provides the legacy richtext wrapper (``<div class="rich-text"></div>``).
```

**Comment by lb-** [comment]
Merged in via a1a2c35c1c1e78efa38bed0074ae0c7968770e84

**Comment by lb-** [comment]
Epic stuff @allcaps  @zerolab  - awesome to see this in

**Pull request closed without merging**